### DISCUSSION TITLE
how to pass python variable to %%sql cell ?

### DISCUSSION THREAD

**Discussion by cometta** [jupyter-incubator/sparkmagic#810] [Q&A]
let say i have value in python variable call  a= 'something'
in 
```
%%sql 
select * from table_name where field = variable_a
```

**Comment by agrounds** [comment]
I don't think this is possible. Instead, I would use regular pyspark to create a temp view matching the condition you need, like so
```python
someDf.where(F.col('field') == x).createTempView('tmp')
```
and then separately run
```
%%sql
select * from tmp
```
